consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -614,45 +740,263 @@ impl PruningProofManager {
 
         drop(pruning_read);
         drop(relations_read);
-        drop(db_lifetime);
+        drop(proof_stores_and_processes.db_lifetime);
+        drop(current_consensus_stores_and_processes.db_lifetime);
 
         Err(PruningImportError::PruningProofNotEnoughHeaders)
     }
 
+    // The "current dag level" is the level right before the level whose parents are
+    // not the same as our header's direct parents
+    //
+    // Find the current DAG level by going through all the parents at each level,
+    // starting from the bottom level and see which is the first level that has
+    // parents that are NOT our current pp_header's direct parents.
+    fn find_current_dag_level(&self, pp_header: &Header) -> BlockLevel {
+        let direct_parents = BlockHashSet::from_iter(pp_header.direct_parents().iter().copied());
+        pp_header
+            .parents_by_level
+            .iter()
+            .enumerate()
+            .skip(1)
+            .find_map(|(level, parents)| {
+                if BlockHashSet::from_iter(parents.iter().copied()) == direct_parents {
+                    None
+                } else {
+                    Some((level - 1) as BlockLevel)
+                }
+            })
+            .unwrap_or(self.max_block_level)
+    }
+
+    fn estimated_blue_depth_at_level_0(&self, level: BlockLevel, level_depth: u64, current_dag_level: BlockLevel) -> u64 {
+        level_depth.checked_shl(level.saturating_sub(current_dag_level) as u32).unwrap_or(level_depth)
+    }
+
+    /// selected parent at level = the parent of the header at the level
+    /// with the highest blue_work (using score as work in this case)
+    fn find_selected_parent_header_at_level(
+        &self,
+        header: &Header,
+        level: BlockLevel,
+        relations_service: &MTRelationsService<DbRelationsStore>,
+    ) -> PruningProofManagerInternalResult<Arc<Header>> {
+        // Parents manager parents_at_level may return parents that aren't in relations_service, so it's important
+        // to filter to include only parents that are in relations_service.
+        let parents = self
+            .parents_manager
+            .parents_at_level(header, level)
+            .iter()
+            .copied()
+            .filter(|parent| relations_service.has(*parent).unwrap())
+            .collect_vec()
+            .push_if_empty(ORIGIN);
+
+        let mut sp = SortableBlock { hash: parents[0], blue_work: self.headers_store.get_blue_score(parents[0]).unwrap_or(0).into() };
+        for parent in parents.iter().copied().skip(1) {
+            let sblock = SortableBlock {
+                hash: parent,
+                blue_work: self
+                    .headers_store
+                    .get_blue_score(parent)
```

Comment:
Here too

Reply:
Fixed in [66d4ebe](https://github.com/kaspanet/rusty-kaspa/pull/494/commits/66d4ebe674f798ec6a2ecc7c16f5e92f9bab8e8f)